### PR TITLE
chore(docs): update main tutorial to incorporate TS Updates to create-gatsby

### PR DESCRIPTION
**Pull request by marvinjude**
## Description

We recently added flags(`-y`, `-ts`) to `create-gatsby`  together with a new prompt  for users to choose between JavaScript and TypeScript. This PR updates docs to reflect these changes and how users can use them. 

## Documentation

https://www.gatsbyjs.com/docs/tutorial/part-1/#create-a-gatsby-site

[sc-44636]

### CODE REVIEW

File: docs/docs/tutorial/part-1/index.mdx [hunk 1]
```diff
@@ -88,6 +88,17 @@ gatsby new
 
 <Announcement style={{marginBottom: "1.5rem"}}>
 
+You can also generate a Gatsby site without needing to go through all the prompts. In this case, the the defaults will be used except for the site name which you'll need to pass.
+You can do this by running the following command from the command line:
+
+```shell:title=Desktop
+ gatsby new <site-name> -y
```
Comment (tyhopp) [inline docs/docs/tutorial/part-1/index.mdx:95]: There is no need for a flag here, `gatsby new`'s interface is different from `npm init gatsby` and `npx create-gatsby`.

Calling [`gatsby new` with a site name](https://www.gatsbyjs.com/docs/reference/gatsby-cli/#new) uses the default starter and skips the questionnaire. Calling `gatsby new` uses the questionnaire.

File: docs/docs/tutorial/part-1/index.mdx [hunk 2]
```diff
@@ -106,9 +117,27 @@ What would you like to name the folder where your site will be created?
...
+```shell
+? Will you be using JavaScript or TypeScript?
+(Single choice) Arrow keys to move, enter to confirm
+  JavaScript
+❯ TypeScript
+```
+
+<Announcement style={{marginBottom: "1.5rem"}}>
+
+This question will be skipped if you passed in a `-ts` flag at the beginning i.e `gatsby new -ts`.
```
Comment (tyhopp) [inline docs/docs/tutorial/part-1/index.mdx:131]: Same here, `-ts` flag is for `npm i gatsby` and `npx create-gatsby`

**Review by tyhopp — CHANGES_REQUESTED** [review]

File: docs/docs/tutorial/part-1/index.mdx [hunk 3]
```diff
@@ -88,6 +88,17 @@ gatsby new
 
 <Announcement style={{marginBottom: "1.5rem"}}>
 
+You can also generate a Gatsby site without needing to go through all the prompts. In this case, the the defaults will be used except for the site name which you'll need to pass.
```
Comment (LekoArts) [inline docs/docs/tutorial/part-1/index.mdx:91]: We should remove this whole part as it's not relevant for the tutorial. We want people to go through all the steps and not bail out early here

File: docs/docs/tutorial/part-1/index.mdx [hunk 4]
```diff
@@ -106,9 +117,27 @@ What would you like to name the folder where your site will be created?
 ✔ Desktop/ my-first-gatsby-site
 ```
 
-5. When the prompt asks, **"Will you be using a CMS?"** select **"No (or I'll add it later)"**.
+5. When the prompt asks, **"Will you be using JavaScript or TypeScript?"** choose between JavaScript or TypeScript.
```
Comment (LekoArts) [inline docs/docs/tutorial/part-1/index.mdx:120]:
```suggestion
5. When the prompt asks, **"Will you be using JavaScript or TypeScript?"** choose JavaScript.
```

File: docs/docs/tutorial/part-1/index.mdx [hunk 5]
```diff
@@ -106,9 +117,27 @@ What would you like to name the folder where your site will be created?
...
+
+```shell
+? Will you be using JavaScript or TypeScript?
+(Single choice) Arrow keys to move, enter to confirm
+  JavaScript
+❯ TypeScript
```
Comment (LekoArts) [inline docs/docs/tutorial/part-1/index.mdx:126]:
```suggestion
❯ JavaScript
  TypeScript
```

File: docs/docs/tutorial/part-1/index.mdx [hunk 6]
```diff
@@ -106,9 +117,27 @@ What would you like to name the folder where your site will be created?
...
+```
+
+<Announcement style={{marginBottom: "1.5rem"}}>
+
+This question will be skipped if you passed in a `-ts` flag at the beginning i.e `gatsby new -ts`.
+This indicates that you want to use TypeScript hence our [Minimal Typescript Starter](https://github.com/gatsbyjs/gatsby/tree/master/starters/gatsby-starter-minimal-ts) will be pulled in as the template.
```
Comment (LekoArts) [inline docs/docs/tutorial/part-1/index.mdx:132]: This announcement should point out that people can use TypeScript but if the want to follow the tutorial step-by-step they need to use JavaScript (as the tutorial uses that)

File: docs/docs/tutorial/part-1/index.mdx [hunk 7]
```diff
@@ -106,9 +117,27 @@ What would you like to name the folder where your site will be created?
...
+Using TypeScript? Also checkout our [TypeScript Docs](/how-to/custom-configuration/typescript/)
+
+</Announcement>
+
+6. When the prompt asks, **"Will you be using a CMS?"** select **"No (or I'll add it later)"**.
 
 ```shell
+
```
Comment (LekoArts) [inline docs/docs/tutorial/part-1/index.mdx:140]:
```suggestion
```

**Review by LekoArts — CHANGES_REQUESTED** [review]

File: docs/docs/tutorial/part-1/index.mdx (hunk 1 above)
Comment (marvinjude) [reply docs/docs/tutorial/part-1/index.mdx]: Thanks! As we earlier discussed, I initially thought we pass the `-y` flag straight into `create-gatsby`

File: docs/docs/tutorial/part-1/index.mdx (hunk 6 above)
Comment (marvinjude) [reply docs/docs/tutorial/part-1/index.mdx]: 👍

File: docs/docs/tutorial/part-1/index.mdx [hunk 8]
```diff
@@ -106,7 +106,22 @@ What would you like to name the folder where your site will be created?
 ✔ Desktop/ my-first-gatsby-site
 ```
 
-5. When the prompt asks, **"Will you be using a CMS?"** select **"No (or I'll add it later)"**.
+5. When the prompt asks, **"Will you be using JavaScript or TypeScript?"** choose JavaScript.
```
Comment (LekoArts) [inline docs/docs/tutorial/part-1/index.mdx:109]:
```suggestion
5. When the prompt asks, **"Will you be using JavaScript or TypeScript?"** choose **JavaScript**.
```

File: docs/docs/tutorial/part-1/index.mdx [hunk 9]
```diff
@@ -106,7 +106,22 @@ What would you like to name the folder where your site will be created?
...
+```shell
+? Will you be using JavaScript or TypeScript?
+(Single choice) Arrow keys to move, enter to confirm
+❯ JavaScript
+  TypeScript
+```
+
+<Announcement style={{marginBottom: "1.5rem"}}>
+
+For this tutorial, we're using JavaScript, feel free to select the TypeScript option if you're interested in using TypeScript instead. Using TypeScript? checkout our [TypeScript Docs](/how-to/custom-configuration/typescript/)
```
Comment (LekoArts) [inline docs/docs/tutorial/part-1/index.mdx:120]:
```suggestion
This tutorial doesn't require any prior TypeScript knowledge as it uses JavaScript. If you're familiar with TypeScript you can read the [Gatsby and TypeScript guide](/how-to/custom-configuration/typescript/) to learn about typings, files, and conventions. If you want to use TypeScript we recommend going through the tutorial first and then only afterwards convert the project to TypeScript.
```

**Review by LekoArts — CHANGES_REQUESTED** [review]

**Review by LekoArts — APPROVED** [review]

**Pull request merged**